### PR TITLE
feature/utterances

### PR DESCRIPTION
**Pull request by sywhb**
- Update wordsCount in page
- Change rate to 1.333
- Read Utterance input

### CODE REVIEW

File: packages/api/src/resolvers/article/index.ts [hunk 1]
```diff
@@ -257,6 +258,9 @@ export const createArticleResolver = authorized<
 
       const saveTime = new Date()
       const slug = generateSlug(parsedContent?.title || croppedPathname)
+      const wordsCount = wordsCounter(parsedContent?.content || '', {
+        isHtml: true,
+      }).wordsCount
```
Comment (jacksonh) [inline packages/api/src/resolvers/article/index.ts:263]: nice. I'm picturing in the future we could have a bunch of metadata on pages that we expose through a REST API, like get /<pageid>/metadata and it just returns key value data for things like word count.

Comment (jacksonh) [reply packages/api/src/resolvers/article/index.ts:263]: I guess one worry is does this function throw on error, or just return zero? We probably don't want to completely fail if an error occurs.

File: packages/api/src/resolvers/article/index.ts [hunk 2]
```diff
@@ -289,6 +293,7 @@ export const createArticleResolver = authorized<
         readingProgressAnchorIndex: 0,
         state: ArticleSavingRequestStatus.Succeeded,
         language: parsedContent?.language,
+        wordsCount,
```
Comment (jacksonh) [inline packages/api/src/resolvers/article/index.ts:296]: So right now we save it but aren't exposing it in GQL, right? (That is fine with me)

File: packages/api/src/resolvers/article/index.ts (hunk 1 above)
Comment (sywhb) [reply packages/api/src/resolvers/article/index.ts:263]: I checked the source code and what it does is convert HTML to text and count words.
It is possible that the converting throws an error so I will write an exception handler to capture any error and return 0 instead.

Yeah, totally agree with the REST API for metadata!

File: packages/api/src/resolvers/article/index.ts (hunk 2 above)
Comment (sywhb) [reply packages/api/src/resolvers/article/index.ts:296]: Yes

File: packages/text-to-speech/src/index.ts [hunk 3]
```diff
@@ -130,41 +149,29 @@ export const textToSpeechStreamingHandler = Sentry.GCPFunction.wrapHttpFunction(
...
-        speechMarksStream,
+        ...utteranceInput,
+        textType: 'utterance',
       }
-      res.set({
-        'Content-Type': 'audio/mpeg',
-        'Transfer-Encoding': 'chunked',
+      const { audioStream, speechMarks } = await synthesizeTextToSpeech(input)
+      res.send({
+        audioData: audioStream.read().toString('hex'),
```
[the remark this answers is not in the document]
Comment (sywhb) [reply packages/text-to-speech/src/index.ts:175]: I will check again. Maybe creating a buffer would be better

**Review by jacksonh — APPROVED** [review]

**Review by jacksonh — APPROVED** [review]

**Pull request merged**